vcd/resource_vcd_nsxv_firewall_rule.go

```diff
@@ -880,6 +880,16 @@ func stringInSlice(str string, list []string) bool {
 	return false
 }
 
+// stringInSlicePartially checks whether the input string contains any of the strings of the slice
+func stringInSlicePartially(str string, list []string) bool {
+	for _, v := range list {
+		if strings.Contains(str, v) {
+			return true
+		}
+	}
+	return false
```

Comment:
```suggestion
	return slices.ContainsFunc(list, func(s string) bool {return strings.Contains(str, s) })
```
It should do the same

Reply:
Thanks for the tip! Changed. It works (test passed).